section3/exercises/arrays.rb

```diff
@@ -18,50 +18,76 @@
 # Store the array in a variable.
 animals = ["Zebra", "Giraffe", "Elephant"]
 print animals
+puts ""
 
 # EXAMPLE: Write code below that will print "Zebra" from the animals array
 # YOUR CODE HERE
-print animals[0]
+puts '-' * 10
+puts animals[0]
 
 # YOU DO: Write code below that will print the number of elements in array of
 # animals from above.
-
+puts ""
+puts animals.count
 
 # YOU DO: Write code that will reassign the last item in the animals
 # array to "Gorilla"
-
-
+puts ""
+animals[2] = "Gorilla"
+puts "position 2 has been changed to #{animals[2]}"
+puts '-' * 10
 # YOU DO: Write code that will add a new animal (type of your choice) to position 3.
-
+animals.push("Fish")
+print animals
+puts ""
+puts '-' * 10
 
 # YOU DO: Write code that will print the String "Elephant" in the animals array
-
-
+puts animals[2]
+puts "the string 'elephant' was replaced by 'Gorilla' as instructed in line 33-37'..."
+puts '-' * 10
 #-------------------
 # PART 2: Foods: Array Methods
 #-------------------
 
 # YOU DO: Declare a variable that will store an an array of at least 4 foods (strings)
 
+foods = ["pizza", "pasta", "burgers", "pudding"]
 
 # YOU DO: Write code below that will print the number of elements in the array of
 # foods from above.
 
+puts "There are #{foods.count} elements in the array 'foods'"
```

Comment:
Good find of `count`! What other methods can we use?

An interesting thing to google before Mod 1 starts...what's the difference various Array methods that return the number of elements? Are any of those "better" than `count` in terms of performance?

^this is a worthwhile rabbit hole to spend ~15 minutes. `count` always works, but a good example to expose you to how methods that seem to do the same thing sometimes work differently under the hood!